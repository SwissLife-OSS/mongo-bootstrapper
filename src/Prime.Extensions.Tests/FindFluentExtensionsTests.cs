using System;
using System.Collections.Generic;
using System.Text;
using MongoDB.Driver;
using Snapshooter.Xunit;
using Squadron;
using System.Threading.Tasks;
using Xunit;

namespace MongoDB.Prime.Extensions.Tests
{
    public class FindFluentExtensionsTests : IClassFixture<MongoResource>
    {
        private readonly IMongoDatabase _mongoDatabase;

        public FindFluentExtensionsTests(MongoResource mongoResource)
        {
            _mongoDatabase = mongoResource.CreateDatabase();
        }

        [Fact]
        public void PrintQuery_PrintOneSingleQuery_OriginalMongoDbQueryPrinted()
        {
            // Arrange
            IMongoCollection<Bar> barCollection =
                _mongoDatabase.GetCollection<Bar>();

            // Act
            string mongodbQuery = barCollection
                .Find<Bar>(bar => bar.Name == "Bar1" || bar.Value == "1234")
                .Limit(5)
                .PrintQuery();

            // Assert
            Snapshot.Match(mongodbQuery);
        }
    }
}
