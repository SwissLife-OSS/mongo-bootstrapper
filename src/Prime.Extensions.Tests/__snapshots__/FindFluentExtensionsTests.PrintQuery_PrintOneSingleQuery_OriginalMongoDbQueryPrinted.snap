find({ "$or" : [{ "Name" : "Bar1" }, { "_id" : "1234" }] }).limit(5)
